modules/data-entry/src/main/frontend/src/questionnaire/Subject.jsx

```diff
@@ -392,8 +392,13 @@ function FormData(props) {
     const questionTitle = entryDefinition["text"];
 
     if (existingQuestionAnswer && existingQuestionAnswer[1]["value"] && (displayed < maxDisplayed)) {
+      // Check for a label and use the label instead of the value, if present
+      let existingQuestionAnswerValue = existingQuestionAnswer[1]["value"];
+      let answerValue = entryDefinition[existingQuestionAnswerValue]["label"]
```

Comment:
I think this may throw an exception, you should use the optional way:
```suggestion
      let answerValue = entryDefinition[existingQuestionAnswerValue]?.["label"]
```